src/cleanvision/utils/viz_manager.py

```diff
@@ -45,6 +45,15 @@ def plot_image_grid(
     fig, axes = plt.subplots(
         nrows, ncols, figsize=(cell_size[0] * ncols, cell_size[1] * nrows)
     )
+
+    for i in range(len(images)):
+        """Converts font size of 7 into inches"""
+        CHARACTER_SIZE_INCHES = 7 * (1 / 72)
+
+        title_width = len(titles[i]) * CHARACTER_SIZE_INCHES
+        if title_width >= cell_size[0]:
+            titles[i] = titles[i][: math.ceil(cell_size[0] / CHARACTER_SIZE_INCHES) - 1]
+
```

Comment:
thanks for the update!  One further improvement IMO will be the following:
After the `if` clause, instead truncate the `titles[i]` a little more than necessary, so that you can append the suffix "..." to the end of the string to indicate it was truncated. 

Please update your main screenshot in PR description with some images where the title is truncated AND some images where the title is not truncated (in the same dataset). With the "..." suffix, it should become more obvious which image paths were truncated vs not.

Reply:
The more advanced version of this that I'd encourage you to consider for this or a future PR, is to do the truncation from the beginning or from the end, whichever makes the truncated strings more unique.  

Currently with the truncation always from the end, you can see there is no way for user to distinguish the file names anymore in your dataset pictured here. When you consider truncating from beginning as well, you can simply add the "..." as a prefix instead of as a suffix.

So for datasets where only the end of the filenames differ, we'd prefer them to be truncated at their beginning.

Reply:
Yes I would be very happy to implement the advanced version. I have started a thread in slack to discuss further on this.

Reply:
Done! Updated in the new commits.

Reply:
thanks! Unit tests are failing though, can you fix and add new unit test for your new functionality?

Reply:
Updated. Handled the case of 1 image in dataset.